lib/msal-core/src/authority/AuthorityFactory.ts

```diff
@@ -25,6 +28,27 @@ export class AuthorityFactory {
         }
     }
 
+    public static async resolveAuthorityAsync(authorityInstance: Authority): Promise<ITenantDiscoveryResponse> {
+        const metadata = await authorityInstance.resolveEndpointsAsync(); // todo: errors?
+        this.metadataMap.set(authorityInstance.CanonicalAuthority, metadata);
+        return metadata;
+    }
+
+    public static parseAuthorityMetadata(authorityUrl: string, authorityMetadataJson: string) {
+        try {
+            if (authorityMetadataJson) {
```

Comment:
Just clarifying: The end user is expected to send a json and we parse out the needed values. And this isn't limited to the three values we are setting here and expandable in the future. Is my understanding correct?

Reply:
Correct, this API expects a JSON string.